src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultStr = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const item = this[i];
+
+      if (item === undefined || item === null) {
+        resultStr += '';
+      } else {
+        resultStr += item;
+      }
```

Comment:
You can simplify your `if else` statement:
```suggestion
      if (item !== undefined && item !== null) {
        resultStr += item;
      }
```

Reply:
Of course, I didn't notice. Thank you!